Warning in clang about unused field -- inlet::Function::m_docEnabled

In our clang@10 config on LC's toss3, there is a warning about `inlet::Function::m_docEnabled` being unused.

```cpp
<axom>/src/axom/inlet/Function.hpp:592:8: error: private field 'm_docEnabled' is not used [-Werror,-Wunused-private-field]
```
Note: This is actually a warning rather an error. I configured with `-DENABLE_WARNINGS_AS_ERRORS=ON`

Is there a plan to use this in the future?
* If not, can we remove this field?
* If so, can we comment it out in the short term?

--- a/src/axom/inlet/Container.cpp
+++ b/src/axom/inlet/Container.cpp
@@ -332,8 +332,7 @@ Function& Container::addFunctionInternal(axom::sidre::Group* sidreGroup,
     fullName,
     cpp11_compat::make_unique<Function>(sidreGroup,
                                         m_sidreRootGroup,
-                                        std::move(func),
-                                        m_docEnabled));
+                                        std::move(func)));
   // emplace_result is a pair whose first element is an iterator to the inserted element
   return *(emplace_result.first->second);
 }

--- a/src/axom/inlet/Function.hpp
+++ b/src/axom/inlet/Function.hpp
@@ -521,11 +521,9 @@ class Function : public Verifiable<Function>
 public:
   Function(axom::sidre::Group* sidreGroup,
            axom::sidre::Group* root,
-           FunctionVariant&& func,
-           bool docEnabled = true)
+           FunctionVariant&& func)
     : m_sidreGroup(sidreGroup)
     , m_sidreRootGroup(root)
-    , m_docEnabled(docEnabled)
     , m_func(std::move(func))
   {
     m_func.setName(name());
@@ -589,7 +587,6 @@ private:
   // This function's sidre group
   axom::sidre::Group* m_sidreGroup = nullptr;
   axom::sidre::Group* m_sidreRootGroup = nullptr;
-  bool m_docEnabled;
   Verifier m_verifier;
   FunctionVariant m_func;
 };

